modules/api/src/main/java/org/eclipse/xpanse/api/controller/EndUserQueryController.java

```diff
@@ -0,0 +1,192 @@
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse/enduser")
+@CrossOrigin
+public class EndUserQueryController {
+
+    @Resource
+    private RegisterService registerService;
+
+    /**
+     * List registered service with query params.
+     *
+     * @param categoryName   name of category.
+     * @param cspName        name of cloud service provider.
+     * @param serviceName    name of registered service.
+     * @param serviceVersion version of registered service.
+     * @return response
+     */
+    @Tag(name = "Deployable Service",
+            description = "APIs to query deployable services.")
+    @Operation(description = "List registered service with query params.")
+    @GetMapping(value = "/register",
+            produces = {MediaType.APPLICATION_JSON_VALUE, "application/hal+json"})
+    @ResponseStatus(HttpStatus.OK)
+    public List<UserDeployableServiceVo> listRegisteredServices(
+            @Parameter(name = "categoryName", description = "category of the service")
+            @RequestParam(name = "categoryName", required = false) String categoryName,
+            @Parameter(name = "cspName", description = "name of the service provider")
+            @RequestParam(name = "cspName", required = false) String cspName,
+            @Parameter(name = "serviceName", description = "name of the service")
+            @RequestParam(name = "serviceName", required = false) String serviceName,
+            @Parameter(name = "serviceVersion", description = "version of the service")
+            @RequestParam(name = "serviceVersion", required = false) String serviceVersion) {
+        RegisteredServiceQuery query = new RegisteredServiceQuery();
+        if (StringUtils.isNotBlank(cspName)) {
+            query.setCsp(Csp.getCspByValue(cspName));
+        }
+        if (StringUtils.isNotBlank(categoryName)) {
+            query.setCategory(Category.getCategoryByCatalog(categoryName));
+        }
+        query.setServiceName(serviceName);
+        query.setServiceVersion(serviceVersion);
+        log.info("List registered service with query model {}", query);
+        List<RegisterServiceEntity> serviceEntities =
+                registerService.queryRegisteredServices(query);
+        String successMsg = String.format("List registered service with query model %s "
+                + "success.", query);
+        List<UserDeployableServiceVo> userDeployableServiceVos =
+                serviceEntities.stream().map(this::convertToRegisteredServiceUserVo)
+                        .collect(Collectors.toList());
+        log.info(successMsg);
+        return userDeployableServiceVos;
+    }
+
+
+    /**
+     * List registered service with category.
+     *
+     * @param categoryName name of category.
+     * @return response
+     */
+    @Tag(name = "Deployable Service",
+            description = "APIs to query deployable services.")
+    @Operation(description = "List registered service group by serviceName, serviceVersion, "
```

Comment:
```suggestion
    @Operation(description = "Get the available services by tree."
```
how about describe it like this?

Reply:
This will be fix in a new PR about 'rename API route'